salt/utils/http.py

```diff
@@ -192,6 +185,36 @@ def query(
     if not backend:
         backend = opts.get("backend", "tornado")
 
+    proxy_host = opts.get("proxy_host", None)
+    if proxy_host:
+        proxy_host = salt.utils.stringutils.to_str(proxy_host)
+    proxy_port = opts.get("proxy_port", None)
+    proxy_username = opts.get("proxy_username", None)
+    if proxy_username:
+        proxy_username = salt.utils.stringutils.to_str(proxy_username)
+    proxy_password = opts.get("proxy_password", None)
+    if proxy_password:
+        proxy_password = salt.utils.stringutils.to_str(proxy_password)
+    no_proxy = opts.get("no_proxy", [])
+
+    if urllib.parse.urlparse(url).hostname in no_proxy:
+        proxy_host = None
+        proxy_port = None
+        proxy_username = None
+        proxy_password = None
+
+    http_proxy_url = None
+    if backend != "requests" and proxy_host and proxy_port:
```

Comment:
Could you move this if statement under the following if statement? Then it would just be
```
if backend != "requests":
```

Reply:
yup, good catch